backend/src/services/memberAttributeSettingsService.ts

```diff
@@ -0,0 +1,161 @@
+import { Attribute } from '../database/attributes/attribute'
+import MemberAttributeSettingsRepository from '../database/repositories/memberAttributeSettingsRepository'
+import SequelizeRepository from '../database/repositories/sequelizeRepository'
+import Error400 from '../errors/Error400'
+import camelCaseNames from '../utils/camelCaseNames'
+import { IServiceOptions } from './IServiceOptions'
+
+export default class MemberAttributeSettingsService {
+  options: IServiceOptions
+
+  constructor(options) {
+    this.options = options
+  }
+
+  async create(data) {
+    const transaction = await SequelizeRepository.createTransaction(this.options.database)
+
+    try {
+      data.name = data.name ?? camelCaseNames(data.label)
+
+      const record = await MemberAttributeSettingsRepository.create(data, {
+        ...this.options,
+        transaction,
+      })
+
+      await SequelizeRepository.commitTransaction(transaction)
+
+      return record
+    } catch (error) {
+      await SequelizeRepository.rollbackTransaction(transaction)
+
+      SequelizeRepository.handleUniqueFieldError(
+        error,
+        this.options.language,
+        'memberAttributeSettings',
+      )
+
+      throw error
+    }
+  }
+
+  async createPredefined(attributes: Attribute[]) {
+    const transaction = await SequelizeRepository.createTransaction(this.options.database)
+
+    try {
+      const created = []
+      for (const attribute of attributes) {
+        // check attribute already exists
+        const existing = await MemberAttributeSettingsRepository.findAndCountAll(
```

Comment:
What happens when we remove an attribute from predefined ones?

Reply:
it'll be very uncommon case (most probably never happened before) so I'm not sure how to handle it, because there'll most likely be data already populated in member.attributes with these fields